### PR TITLE
Exam by Voytkiv Vlad IM-33

### PR DESCRIPTION
**Pull request by alomj**

### CODE REVIEW

**Comment by DavidkJun** [comment]
after the review i can agree with all the commits

**Comment by Kiryaoo** [comment]
Agree with all commits

**Comment by Adaptolog** [comment]
It looks solid overall

**Comment by yur-ochka** [comment]
I reviewed all commits and can agree with all of them!

**Comment by nikama-s** [comment]
I reviewed all changes and totally agree with them

**Comment by namanica** [comment]
good job has been done!

**Comment by thonly-zukich** [comment]
Great work on GitHub! Your code is clean, well-organized, and easy to understand. Keep it up!

**Comment by D-Vasylchenko** [comment]
Good job with code, all clean and organized

**Comment by gbdfxilhn** [comment]
that's solid work. well done

File: Tasks/between.js [hunk 1]
```diff
@@ -1,21 +1,20 @@
...
-getvaluebetween = (str, p, s) => {
-  i = str.indexOf(p);
-  if (i === -1) return '';
+// Step 1: Renaming variables
+// Step 2: Adding const`s and let
+// Step 3: Restiling
+const getValueBetween = (fullString, prefix, suffix) => {
+  let index = fullString.indexOf(prefix);
+  if (index === -1) return '';  
   else {
```
Comment (tshemsedinov) [inline Tasks/between.js:8]: Remove else

Comment (tshemsedinov) [inline Tasks/between.js:8]: Remove else

File: Tasks/between.js [hunk 2]
```diff
@@ -1,21 +1,20 @@
...
-    k = i + p.length;
-    str = str.substring(k);
-    if (s) {
-      i = str.indexOf(s);
-      if (i === -1) {
-        return '';
-      } else {
-        str = str.substring(0, i);
-      }
+    let prefixLength = index + prefix.length;
```
Comment (tshemsedinov) [inline Tasks/between.js:9]: const

File: Tasks/between.js [hunk 3]
```diff
@@ -1,21 +1,20 @@
...
+    fullString = fullString.substring(prefixLength);
+    if (suffix) {
+      index = fullString.indexOf(suffix);
```
Comment (tshemsedinov) [inline Tasks/between.js:12]: Use new identifiers, it's free

File: Tasks/between.js [hunk 4]
```diff
@@ -1,21 +1,20 @@
...
+      if (index === -1)  return '';
+      else fullString = fullString.substring(0, index);
     }
   }
-  return str;
+  return fullString;
```
Comment (tshemsedinov) [inline Tasks/between.js:17]: Don't change incoming args

File: Tasks/day.js [hunk 5]
```diff
@@ -1,15 +1,16 @@
...
 
-let D = ['Sun', 'Mon', 'Tue', 'Wed', 'Thu', 'Fri', 'Sat'];
-
-_parse_day_ = (s) => {
-  let i;
-  for (i = 0; i < D.length; i++) {
-    if (s.startsWith(D[i].toLowerCase())) {
+const numDay = (dayString) => {
+  for (let i = 0; i < daysOfWeek.length; i++) {
+    if (dayString.startsWith(daysOfWeek[i].toLowerCase())) {
```
Comment (tshemsedinov) [inline Tasks/day.js:9]: Use intermediate identifiers

File: Tasks/filter.js [hunk 6]
```diff
@@ -1,15 +1,25 @@
...
   }
-  for (x of remove) T.splice(x, 1);
-  return T;
+
+  for (const element of remove){
+    array.splice(element, 1);
+
+  }
+
+  return array;
```
Comment (tshemsedinov) [inline Tasks/filter.js:22]: Do not change incoming args

File: Tasks/intersection.js [hunk 7]
```diff
@@ -1,25 +1,18 @@
...
+// Step 1: Renaming variables
+// Step 2: I have added const to my function and "element"
+// Step 3: Making code more readable
+// Step 4: I`ve used  not-equal-to operator to provide cleaner code
+const intersection = function intersection(firstDictionary, secondDictionary) {
+  keys = Object.keys(firstDictionary);
+  
+  for (const element of keys) {
+    if (firstDictionary[element] !== secondDictionary[element]) {
+       delete firstDictionary[element];
```
Comment (tshemsedinov) [inline Tasks/intersection.js:11]: Implement without delete

File: Tasks/ip.js [hunk 8]
```diff
@@ -1,19 +1,27 @@
...
+// Step 1: renamed variables
+// Step 2: added const`s and let
+// Step 3: restiling code and make him more readable
+const parseip = (input) => {
+const array = [];
+  if (input === ''){ 
+     return;
+  }  else {
+  const arrayOfIpEl = input.split('.');
+    if (arrayOfIpEl.length !== 4){
```
Comment (tshemsedinov) [inline Tasks/ip.js:11]: Magic number

File: Tasks/ip.js [hunk 9]
```diff
@@ -1,19 +1,27 @@
...
-      a[j] = parseInt(b);
-      if (isNaN(a[j])) return;
-      j++;
+// Step 1: renamed variables
+// Step 2: added const`s and let
+// Step 3: restiling code and make him more readable
+const parseip = (input) => {
+const array = [];
+  if (input === ''){ 
+     return;
```
Comment (tshemsedinov) [inline Tasks/ip.js:8]: Wat?

File: Tasks/ip.js [hunk 10]
```diff
@@ -1,19 +1,27 @@
...
+     
+      index++;
     }
   }
-  return a;
+  return array;
 };
 
-module.exports = Parseip;
+module.exports = parseip;
```
Comment (tshemsedinov) [inline Tasks/ip.js:27]: Terrible formatting

File: Tasks/month.js [hunk 11]
```diff
@@ -15,12 +16,13 @@ Months = [
...
 
-Month = (s) => {
-  l = Months.length;
-  for (let i = 0; i < l; i++) {
-    if (s.toLowerCase().startsWith(Months[i])) return i + 1;
+
+const monthNumber = (givenMonth) => {
+
+  for (let i = 0; i < months.length; i++) {
+    if (givenMonth.toLowerCase().startsWith(months[i])) return i + 1;
```
Comment (tshemsedinov) [inline Tasks/month.js:23]: Too long expression

File: Tasks/password.js [hunk 12]
```diff
@@ -1,13 +1,15 @@
...
+//Step 1: Renamed function from GeneratePaswword to generatePassword. Renamed variables
+//Step 2: Used const instead of let for the function declaration to indicate that the function reference will not be reassigned.
+//Step 3: Used addition assignment for better code readability
+const generatePassword = (alphabet, length) => {
+  const maxPassLen = alphabet.length;
+  pasw = '';
   for (let i = 0; i < length; i++) {
-    Index = Math.floor(Math.random() * MAX);
-    key = key + alphabet[Index];
+    Index = Math.floor(Math.random() * maxPassLen);
```
Comment (tshemsedinov) [inline Tasks/password.js:9]: Not declared

**Pull request closed without merging**